app/src/main/java/com/google/droidjump/leveldata/LevelData.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump.leveldata;
+
+import android.content.res.Resources;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedList;
+import java.util.Scanner;
+
+public class LevelData {
+    private int baseSpeed;
+    private LinkedList<Integer> intervals;
+    private LinkedList<String> obstacleTypes;
+
+    public LevelData(Level level, Resources resources) {
+        intervals = new LinkedList<>();
```

Comment:
I would suggest to unite interval and obstacle type in one class named "ObstacleData" or something alike. Right now you have two lists which should be read in parallel using one index which is error prone.

Reply:
Done